clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -3783,10 +3783,19 @@ LValue CodeGenFunction::EmitArraySubscriptExpr(const ArraySubscriptExpr *E,
     Addr = EmitPointerWithAlignment(E->getBase(), &EltBaseInfo, &EltTBAAInfo);
     auto *Idx = EmitIdxAfterBase(/*Promote*/true);
     QualType ptrType = E->getBase()->getType();
+
+    const ValueDecl *PtrDecl = nullptr;
+    if (const auto *DRE =
+            dyn_cast<DeclRefExpr>(E->getBase()->IgnoreParenCasts()))
+      PtrDecl = DRE->getDecl();
+    else if (const auto *ME =
+                 dyn_cast<MemberExpr>(E->getBase()->IgnoreParenCasts()))
+      PtrDecl = ME->getMemberDecl();
+
     Addr = emitArraySubscriptGEP(*this, Addr, Idx, E->getType(),
                                  !getLangOpts().isSignedOverflowDefined(),
                                  SignedIndices, E->getExprLoc(), &ptrType,
-                                 E->getBase());
+                                 E->getBase(), "ptridx", PtrDecl);
```

Comment:
Is this name expected to appear in the output?  Otherwise, LGTM.

Reply:
Yes, its basically the 'hint' for the name of the GEP in IR.  That said, those are often stripped in non-debug builds :)